### PR TITLE
experimental: add package and move recv buffer pool APIs into it

### PR DESCRIPTION
**Pull request by easwars**
RELEASE NOTES: none

### CODE REVIEW

**Review by dfawley — COMMENTED** [review]
Do you have a dependent PR that builds on this by adding/moving things here?  Can we include this with that?  It doesn't seem very useful to have an experimental package with nothing in it.

**Comment by easwars** [comment]
> Do you have a dependent PR that builds on this by adding/moving things here? Can we include this with that? It doesn't seem very useful to have an experimental package with nothing in it.

Added the recv buffer pool APIs to the experimental package. PTAL. Thanks.

File: dialoptions.go [hunk 1]
```diff
@@ -705,12 +706,16 @@ func WithIdleTimeout(d time.Duration) DialOption {
...
-// # Experimental
-//
-// Notice: This API is EXPERIMENTAL and may be changed or removed in a
-// later release.
+// Deprecated: use experimental.WithRecvBufferPool instead.  Will be deleted in
+// v1.60.0 or later.
 func WithRecvBufferPool(bufferPool SharedBufferPool) DialOption {
 	return newFuncDialOption(func(o *dialOptions) {
 		o.recvBufferPool = bufferPool
 	})
```
Comment (dfawley) [inline dialoptions.go:714]: `return withRecvBufferPool(bufferPool)`

File: experimental/experimental.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package experimental is a placeholder for experimental features that might
```
Comment (dfawley) [inline experimental/experimental.go:19]: s/placeholder for/ collection of/

**Pull request merged**